helper/backoff.go

```diff
@@ -0,0 +1,18 @@
+package helper
+
+import "time"
+
+// ExponentialBackoff performs exponential backoff attempts on a given action
+func ExponentialBackoff(action func() error, max int, initial time.Duration) error {
```

Comment:
@JekaMas would love your insight over here

Reply:
This is only one function, I can just move it to the package where it is used. But decided to move it to a separate package to be more accurate. There should be a balance between redundant dependencies and creating simple functions by us.

Reply:
In this case, we should remove all functions from utils and import third-party libs instead of all of them.

Reply:
If we introduce something like that it should be at least well tested.
From my point of view, I'd stick to solutions that don't increase tech debt. So could we check if there're any well known and tested functions as 3rd party libs and if there're any then I'd rather use that library than have not tested backoff. Introducing tests is also a solution.

For me personally the quality of the code is not being measured by number of dependancies.

Reply:
There is just one simple function that is covered by tests. It completely does not make sense to use 3rd party for this IMO.

If you think we need to use 3rd party for this, then how come we don't use it for all helper/until functions? 

> For me personally the quality of the code is not being measured by a number of dependencies.
Until there are a lot of dependencies and one of them triggers conflicts between versions. I had such an experience before.

Reply:
Regarding testing, this simple function has enough test coverage and we can fully control it. If we use 3rd party, the ownership of this lib is under someone who does not have any responsibilities, and we can't be sure it is fully tested and maintained. I'm fine with using already existing solutions for relatively complex tools, but not just for a function with a few lines of code.

Reply:
> The function is not covered by tests

No way, just a notifier I didn't commit the file with tests of this function. Just pushed.

Agree. My point is that adding new dependencies instead of creating a new function with a few lines of code is a little bit... redundant I would say. Just went over the code and saw a lot of helper functions. Should we add 3rd party libs for them as well? Or just for this function? If just for this one, how to understand if we need 3rd party or not for a specific function? If I just put this function into `listener` package, would it be ok or not? Or if I just remove it at all?